### PR TITLE
Add cluster-type and related config

### PR DESCRIPTION
**Pull request by wangdian**
Use [service name]_[cluster type].yaml file for default config for cluster type.
If not found cluster type config file, use [service name].yaml instead

### CODE REVIEW

**Comment by ydye** [comment]
why not add this type into cluster.coomon?

**Comment by wangdian** [comment]
The cluster-type is in cluster.common

cluster:
  common:
      cluster-id: pai_ut
      cluster-type: yarn

File: src/cluster/config/cluster.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def merge_service_configuration(self, overwrite_srv_cfg, default_srv_cfg):
     def validation_common(self, common_configuration):
         if "cluster-id" not in common_configuration:
             return False, "cluster-id is miss in service-configuration.yaml -> cluster -> common -> cluster-id"
+        if "cluster-type" not in common_configuration or common_configuration["cluster-type"] not in ["yarn", "k8s"]:
```
Comment (abuccts) [inline src/cluster/config/cluster.py]: does `cluster-type` has a default value for the previous config without this field?

File: src/rest-server/config/rest-server_k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation
```
Comment (abuccts) [inline src/rest-server/config/rest-server_k8s.yaml]: use `.k8s.yaml` in filename

File: src/webportal/config/webportal_k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) Microsoft Corporation
```
Comment (abuccts) [inline src/webportal/config/webportal_k8s.yaml]: is this file included in this PR?

File: src/rest-server/config/rest-server_k8s.yaml (hunk 2 above)
Comment (wangdian) [reply src/rest-server/config/rest-server_k8s.yaml]: Ok

File: src/webportal/config/webportal_k8s.yaml (hunk 3 above)
Comment (wangdian) [reply src/webportal/config/webportal_k8s.yaml]: Yes, will change to .k8s.yaml.

File: src/cluster/config/cluster.py (hunk 1 above)
Comment (wangdian) [reply src/cluster/config/cluster.py]: The previous config does not have cluster-type.  Please refer to forward_compatibility.py, when use old config file, use "yarn" as default cluster-type

**Pull request merged**